homeassistant/components/smlight/update.py

```diff
@@ -178,7 +178,7 @@ def _update_finished(self, event: MessageEvent) -> None:
     @callback
     def _update_failed(self, event: MessageEvent) -> None:
         self._update_done()
-
+        self.coordinator.in_progress = False
```

Comment:
Probably not for this PR, but we only seem to set this to True in the `async_install` method. If an update is initiated from outside of Home Assistant, can we detect that?

Reply:
> If an update is initiated from outside of Home Assistant, can we detect that?

Yes, If an update is initiated from the Smlight web UI, we would still receive the same set of events: progress, done and/or fail indicating progress. So it should be possible to leave the event callbacks always in place and detect update from outside HA.